internal/kafka/internal/workers/clusters_mgr.go

```diff
@@ -190,18 +190,20 @@ func (c *ClusterManager) Reconcile() []error {
 }
 
 func (c *ClusterManager) processMetrics() []error {
+	var errs []error
 	if err := c.setClusterStatusCountMetrics(); err != nil {
-		return []error{errors.Wrapf(err, "failed to set cluster status count metrics")}
+		errs = append(errs, errors.Wrapf(err, "failed to set cluster status count metrics"))
 	}
 
 	if err := c.setKafkaPerClusterCountMetrics(); err != nil {
-		return []error{errors.Wrapf(err, "failed to set kafka per cluster count metrics")}
+		errs = append(errs, errors.Wrapf(err, "failed to set kafka per cluster count metrics"))
 	}
 
 	if err := c.setClusterProviderResourceQuotaMetrics(); err != nil {
```

Comment:
Question around this:

This method iterates on all QuotaCost of the organization that the AMS user account making the request belongs to.

If you take a look for example to the AMS stage organization we are using the results include non-rhosak QuotaCosts too like for example:
```
    {
      "allowed": 33024,
      "consumed": 24,
      "href": "/api/accounts_mgmt/v1/organizations/1gxCqJ5UorERrM42S8fRz8lT3ML/quota_cost",
      "kind": "QuotaCost",
      "organization_id": "1gxCqJ5UorERrM42S8fRz8lT3ML",
      "quota_id": "compute.node|cpu|byoc|osd",
      "version": "1ee76705-de4f-4c60-a39c-1c914719dd6c"
    },
```

or

```
    {
      "allowed": 12,
      "consumed": 0,
      "href": "/api/accounts_mgmt/v1/organizations/1gxCqJ5UorERrM42S8fRz8lT3ML/quota_cost",
      "kind": "QuotaCost",
      "organization_id": "1gxCqJ5UorERrM42S8fRz8lT3ML",
      "quota_id": "cluster|rhinfra|rhosr|any",
      "version": "1ee76705-de4f-4c60-a39c-1c914719dd6c"
    },
```

No filtering of any kind is being performed. It seems to me that is not what we really want. Is that intentional?

Reply:
it's intentional to avoid hardcoding values for filtering these quotas. It was discussed here: https://issues.redhat.com/browse/MGDSTRM-9403

Reply:
A while ago I implemented the `GetQuotaCostsForProduct` method where you can filter by the product name and resource name and you don't need to filter specifying the entire quota IDs.
The logic is based on filtering based on the RelatedResources in AMS.
Maybe we could reuse that to get the information for the products and resource names we are interested on?

Reply:
We could. We may also need to include `resource_type` in the filters as the compute nodes returns ~300 related resources.  We'll need to hardcode the following in the ocm provider: product name ('OSD'/'ANY'), resource_name (addon names) and resource_type ('add-on', 'compute.node', 'cluster'). One other thing to note is we'll need to update these in case the resource information changes/a new quota has been added, which doesn't really happen often.

Does the above sound good to you @miguelsorianod @machi1990 @pb82?

Reply:
We would need to check if it is possible to filter at ReservedResource level on AMS API server side.

I am not too worried about updating when changes/new quota has been added. It shouldn't happen often: we don't add new addons or products frequently

Reply:
Not sure if there is a way. I had a look at the api docs for that [endpoint](https://api.openshift.com/?urls.primaryName=Accounts%20management%20service#/default/get_api_accounts_mgmt_v1_organizations__orgId__quota_cost), there is a search parameter but I don't think it's possible to use it on the `related_resources` column since that's an array. We may have to filter the results on our side.

Reply:
I've updated this PR with the proposed changes in this thread, see https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1251/commits/7294f6f31ecbd2d1c0b719f4b0956efc1f596c45. Please take a look when you get a chance.

Reply:
the new commit lgtm.

I'll let @miguelsorianod have a look as well